src/requestVariableCompletionItemProvider.ts

```diff
@@ -0,0 +1,97 @@
+'use strict';
+
+import { CompletionItemProvider, TextDocument, Position, CancellationToken, CompletionItem, CompletionItemKind, Range, TextLine } from 'vscode';
+
+import { RequestVariableCacheValueProcessor } from "./requestVariableCacheValueProcessor";
+import { VariableProcessor } from "./variableProcessor";
+import { VariableUtility } from "./variableUtility";
+
+const firstPartRegex: RegExp = /^(\w+)\.$/;
+const secondPartRegex: RegExp = /^(\w+)\.(request|response)\.$/;
+
+export class RequestVariableCompletionItemProvider implements CompletionItemProvider {

```

Comment:
Do we need to only allow these completion items when the first part or so called _variable name_ is a request variable?

Reply:
You mean checking if the request variable exists at all in memory? I'm not sure that is developer friendly, personally I like having completion on "known" valid syntax, even if it won't run.

Is there a good reason to do that?

Reply:
@cbrevik  I don't express myself clearly, I just mean this completion item hint only make sense for request variable, not for other types of variables. So it's not necessary for request variables loaded in memory, just exist or already declared in file is enough.

Reply:
I was thinking maybe the check at the top was enough;
```typescript
if (!VariableUtility.isPartialRequestVariableReference(document, position)) {
    return [];
}
```

Which basically checks if the path is using a dot, which would by definition would be a request variable. (Since only that type uses dotted values).

But it does not check if the variable name is declared in the file as a request variable, is that what you're thinking should also be checked?

Reply:
I've added a check for if the variable name is defined in the file as a request variable (and not necessarily in memory)